integrations/gitlab/gitlab_integration/ocean.py

```diff
@@ -207,3 +212,35 @@ async def resync_pipelines(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
                         {**pipeline.asdict(), "__project": project.asdict()}
                         for pipeline in pipelines_batch
                     ]
+
+
+@ocean.on_resync(ObjectKind.MEMBER)
+async def resync_members(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    gitlab_resource_config: GitlabMembersResourceConfig = typing.cast(
+        GitlabMembersResourceConfig, event.resource_config
+    )
+    selector = gitlab_resource_config.selector
+
+    async def fetch_group_members(service, group):
+        if selector.enrich_with_public_email:
+            enriched_member_tasks = [
+                service.enrich_member_with_public_email(member)
+                async for members in service.get_all_group_members(group)
+                for member in members
+            ]
+            enriched_members = await asyncio.gather(*enriched_member_tasks)
+            return enriched_members
+        else:
+            member_dicts = [
+                member.asdict()
+                async for members in service.get_all_group_members(group)
+                for member in members
+            ]
+            return member_dicts
+
+    for service in get_cached_all_services():
+        async for groups in service.get_all_groups(skip_validation=True):
+            group_tasks = [fetch_group_members(service, group) for group in groups]
+            for group_task in asyncio.as_completed(group_tasks):
+                group_members = await group_task
+                yield group_members
```

Comment:
I am confused which is one is the right one? you both return members_dics and the group_members in the same resync method? how should it work?

Reply:
Members dict is returned by the `fetch_group_members` function which is responsible for making the decision as to whether to enrich or not enrich a group member #

Reply:
at `group_tasks = [fetch_group_members(service, group) for group in groups]`, we leverage `fetch_group_members` to fetch all members of the each group and process them as completed.